lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,311 @@
+
+      className = classnames('nx-combobox', classNameProp, {
+        'nx-combobox--dropdown-showable': showDropdown || showAlert
+      }),
+      alertClassName = classnames('nx-combobox__alert', {
+        'nx-combobox__alert--error': !!loadError
+      }),
+      inputDescribedby = classnames(ariaDescribedBy, {
+        [alertDropdownId]: showAlert
+      });
+
+  // There is a requirement that when there is an error querying the data, if the user navigates away from
+  // the component and then comes back to it the search should be retried automatically
+  function handleComponentFocus(evt: FocusEvent<HTMLDivElement>) {
+
+    if (loadError) {
+      // check if this is focus coming into the component from somewhere else on the page, not just moving between
+      // children of this component and not from focus coming into the browser from some other window
+      const comingFromOutsidePage = evt.relatedTarget === null,
+          comingFromChildNode = evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget);
+
+      if (!(comingFromOutsidePage || comingFromChildNode)) {
+        doSearch(value);
+      }
+    }
+  }
+
+  function handleComponentBlur(evt: FocusEvent<HTMLDivElement>) {
+
+    if (!(evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget))) {
+      // The automatically selected suggestion becomes the value of the combobox
+      // when the combobox loses focus.
+      if (autoComplete && focusableBtnIndex !== null) {
+        const elToFocusText = matches[focusableBtnIndex].displayName;
+        onChange(elToFocusText);
+        onSearch(elToFocusText.trim());
+      }
+      setFocusableBtnIndex(null);
+      setInlineStyle(false);
+    }
+  }
+
+  function getDropdownBtnIdForIndex(idx: number) {
+    return `${dropdownBtnIdPrefix}-${idx}`;
+  }
+
+  function handleOnChange(newVal: string) {
+    setFocusableBtnIndex(null);
+    onChange(newVal);
+
+    if (newVal.trim() !== value.trim()) {
+      doSearch(newVal);
+    }
+  }
+
+  function doSearch(val: string) {
+    focusTextInput();
+    onSearch(val.trim());
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+
+        if (elToFocus) {
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          elToFocus.scrollIntoView({ block: 'nearest' });
+          setInlineStyle(false);
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleOnClick({displayName}: DataItem<T, string>) {
```

Comment:
Can this be called `handleDropdownBtnClick` in order to make its use more clear?

Reply:
Yes, updated here b560f99f1ec3229b35d38153df79873c88368dfc.